include/nl-priv-dynamic-core/hashtable-api.h

```diff
@@ -0,0 +1,37 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/*
+ * Private resizable hash table API for libnl.
+ *
+ * The public header <netlink/hashtable.h> exposes the fixed-size hash table
+ * helpers.  The resizable hash table helpers which were recently introduced
+ * in lib/hashtable.c are intended for internal use only and therefore live
+ * in the private header namespace below include/nl-priv-dynamic-core/.
+ */
+
+#ifndef NETLINK_HASHTABLE_API_H_
+#define NETLINK_HASHTABLE_API_H_
+
+#include "nl-priv-dynamic-core/object-api.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
There is no need to have `extern "C"` here. 

This header is internal-only (only used to build libnl3 itself). Users must build libnl3 with a C compiler. And since other such headers also don't have `extern "C"`, you anyway couldn't use a C++ compiler to build libnl3.